lib/udf_json_dicts.rb

```diff
@@ -0,0 +1,46 @@
+class UdfJsonDicts
+  UDFS = [
+      {
+          type:        :function,
+          name:        :json_extract_path_keys,
+          description: "Return the list of keys in a json dict as a json string",
+          params:      "jsonstr varchar(max)",
+          return_type: "varchar(max)",
+          body:        %~
+            import json
+            if not jsonstr:
+              return '[]'
+            else:
+              return json.dumps(sorted(json.loads(str(jsonstr)).keys()))
```

Comment:
I was referring specifically to sorting of the results, I guess you can demand in some way that all input for this function are sorted, this way redshift takes care of sorting it instead of python, Although I am not sure there's even a difference hence me questioning if these will be "pushed down". Finally this blog post https://medium.com/wunderlist-engineering/amazon-redshift-s-udf-b3d4205ae732#.myzb5v9dh

Made realise the conditions are not being pushed down so if we have performance issues we where we can start.